### ISSUE TITLE
MonthDaySerializer does not serialize objects into array

### ISSUE THREAD

**Issue by XhstormR** [FasterXML/jackson-modules-java8#107]
Unlike YearMonthSerializer serialize objects as  array " [2018, 5]", MonthDaySerializer serialize to "--03-02"

I notice [document](http://fasterxml.github.io/jackson-datatype-jsr310/javadoc/2.8/com/fasterxml/jackson/datatype/jsr310/ser/MonthDaySerializer.html) say, what should i do :
```
NOTE: unlike many other date/time type serializers, this serializer will only use Array notation if explicitly instructed to do so with JsonFormat (either directly or through per-type defaults) and NOT with global defaults.
```

I use org.springframework.boot:spring-boot-starter-web:2.1.3.RELEASE,
com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.9.8,
com.fasterxml.jackson.module:jackson-module-kotlin:2.9.8

application.yml
```
  jackson:
    serialization:
      INDENT_OUTPUT: false
      WRITE_DATES_AS_TIMESTAMPS: true
```

**Comment by cowtowncoder** [comment]
`MonthDaySerializer` outputs things as JSON array by default (looking at code), so I would need a reproduction of your issue. It could be due to framework you are using ignoring `WRITE_DATES_AS_TIMESTAMPS` setting.

**Comment by XhstormR** [comment]
Thanks, this is reproduction.

[TEST1.zip](https://github.com/FasterXML/jackson-modules-java8/files/3011819/TEST1.zip)

```
C:\Users\Leo>curl http://127.0.0.1:8080/ -i
HTTP/1.1 200
Content-Type: application/json;charset=UTF-8
Transfer-Encoding: chunked
Date: Wed, 27 Mar 2019 06:26:49 GMT

"--03-27"
```

**Comment by wilkinsona** [comment]
From the referenced Spring Boot issue, here's an even smaller reproduction:

```java
ObjectMapper objectMapper = new ObjectMapper().enable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS).registerModule(new JavaTimeModule());
System.out.println(objectMapper.writeValueAsString(MonthDay.now()));
System.out.println(objectMapper.writeValueAsString(YearMonth.now()));
System.out.println(objectMapper.writeValueAsString(LocalDateTime.now()));
```

The above code will produce output similar to the following:

```
"--03-28"
[2019,3]
[2019,3,28,15,35,59,579000000]
```

**Comment by XhstormR** [comment]
Has anyone noticed here?

**Comment by cowtowncoder** [comment]
Ok finally got around to this one.

So: for some reason, original implementation defaulted to textual, not array-based, and since there is no way to distinguish "left as default" setting for use-timestamps from explicit, indication has to be done some other way.
There are 2 main mechanisms: annotation with `@JsonFormat` and "config overrides".

Former is

```java
    static class Wrapper {
        @JsonFormat(shape = JsonFormat.Shape.ARRAY) // or NUMBER
        public MonthDay value;
    }
```

and config overrides something like

```java
objectMapper.configOverride(MonthDay.class)
    .setFormat(JsonFormat.Value.forShape(JsonFormat.Shape.ARRAY);
```

Now: these work, but it seems that deserialization part does NOT actually work at this point.
So I'll leave the issue open for now. I will add tests for serialization for 2.9, but deserialization fix probably needs to go in 2.10.0.

**Comment by cowtowncoder** [comment]
Closing this, follow-up under #121.
